index.js

```diff
@@ -185,10 +185,6 @@ HtmlWebpackPlugin.prototype.evaluateCompilationResult = function(compilation, co
   if(!compilationResult) {
     return Promise.reject('The child compilation didn\'t provide a result');
   }
-  if(typeof source === 'string') {
```

Comment:
this part allows to pass text html files instead of templates - why would you like to remove it?